server/src/main/java/keywhiz/auth/mutualssl/SpiffePrincipal.java

```diff
@@ -0,0 +1,28 @@
+package keywhiz.auth.mutualssl;
+
+import java.net.URI;
+import java.security.Principal;
+
+public class SpiffePrincipal implements Principal {
+  private final URI spiffeId;
+
+  public SpiffePrincipal(URI spiffeId) {
+    this.spiffeId = spiffeId;
+  }
+
+  @Override public String getName() {
+    return spiffeId.toString();
+  }
+
+  public URI getSpiffeId() {
+    return spiffeId;
+  }
+
+  /**
+   * Use the workload id of a Spiffe Id as the client name.
+   */
+  public String getClientName() {
+    String path = spiffeId.getPath();
+    return path.isEmpty() ? path : path.substring(1);
```

Comment:
Why do we drop the first character from the `spiffeId` path here, with `path.substring(1)`?

Reply:
To drop the '/' character